src/platform/intel/ace/platform.c

```diff
@@ -86,16 +86,15 @@ static struct pm_notifier pm_state_notifier = {
 /* Runs on the primary core only */
 int platform_init(struct sof *sof)
 {
-	int ret;
+	int ret = 0;
 
 	trace_point(TRACE_BOOT_PLATFORM_CLOCK);
 	platform_clock_init(sof);
-	kcps_budget_init();
 
 #if CONFIG_KCPS_DYNAMIC_CLOCK_CONTROL
+	kcps_budget_init();
+
 	ret = core_kcps_adjust(cpu_get_id(), PRIMARY_CORE_BASE_CPS_USAGE);
-#else
```

Comment:
@abonislawski @serhiy-katsyuba-intel Can you double-check this? It seems unnecessary to set the DSP frequency from SOF side if there is no dynamic control (the Zephyr platform code should choose the default speed).

Reply:
@abonislawski any comment?